src/magic_ter_fur_transform.cpp

```diff
@@ -0,0 +1,287 @@
+#include "magic.h"
```

Comment:
There is no "magic_ter_fur_transform.h" header for this cpp file. Where is the class definition for ` ter_furn_transform`. I see, it's in "magic.h". This is confusing. I suggest you move the class and its implementation into a single cpp/h file pair. Or move the content of this cpp file into "magic.cpp".

The current system is just confusing. If I see a cpp file containing the functions for a class, I expect definition to be in the matching header file. And if I see a class definition in a header file, I expect the functions to be in the matching cpp file.